nep-Y.mediawiki

```diff
@@ -0,0 +1,324 @@
+long as they fit the purpose.
+
+{|
+|+ Error code ranges
+|-
+! Range           !! Usage
+|-
+| -15000...-10000 || Experimental implementation-defined range.
+|-
+| -699...-600     || Configuration and service errors.
+|-
+| -599...-500     || Inventory verification errors.
+|-
+| -399...-300     || Wallet-related errors.
+|-
+| -199...-100     || Missing/unknown items/inventories/states.
+|}
+
+Experimental codes can be used by node implementations if they're documented,
+however clients can't rely on the meaning of these codes if the version of the
+node changes or if they connect to a different node implementation. New
+proposals can set new ranges or specific codes as long as they don't conflict
+with this specification. All the other codes are reserved for future use and
+MUST NOT be used by compliant implementations.
+
+====Missing items====
+
+{|
+|+ Codes for missing items
+|-
+! Code !! Message            !! Meaning
+|-
+| -101 || Unknown block      || Call that accepts as a parameter or searches for a header or a block as a part of its job can't find it.
+|-
+| -102 || Unknown contract   || Call that accepts as a parameter or searches for a contract as a part of its job can't find it.
+|-
+| -103 || Unknown transaction|| Call that accepts as a parameter or searches for a transaction as a part of its job can't find it.
+|-
+| -104 || Unknown storage item|| Call that looks for an item in the storage can't find it.
+|-
+| -105 || Unknown script containter|| Call that accepts as a parameter or searches for a script container (a block or transaction) as a part of its job can't find it (this error generalizes -101 and -103 in cases where it's needed).
+|-
+| -106 || Unknown state root || Call that accepts as a parameter or searches for a state root as a part of its job can't find it.
+|-
+| -107 || Unknown session    || Call that accepts as a parameter or searches for a session as a part of its job can't find it.
+|-
+| -108 || Unknown iterator   || Call that accepts as a parameter or searches for a session iterator as a part of its job can't find it.
+|-
+| -109 || Unknown height     || Block or header height passed as parameter or calculated during call execution is not correct (out of the range known to the node).
+|}
+
+
+====Wallet-related problems====
+
+{|
+|+ Codes for calls that use a wallet
+|-
+! Code !! Message            !! Meaning
+|-
+| -300 || Insufficient funds || Transaction that sends some assets can't be created because it fails.
+|-
+| -301 || Fee limit exceeded || Transaction requires more network fee to be paid than is allowed by settings.
+|-
+| -302 || No opened wallet   || Server doesn't have any opened wallet to operate with.
+|-
+| -303 || Wallet not found   || Specified (or configured) wallet file path is invalid.
+|-
+| -304 || Wallet not supported|| Specified (or configured) file can't be opened as a wallet.
+|}
+
+====Verification errors====
+
+{|
+|+ Inventory verification or verification script errors
```

Comment:
> Either negative

`FormatException`.

> `MaxVerificationGas`

Hmm, that's for the network fee and it's not an error to have it more than `MaxVerificationGas`, it's just that only this portion of the fee will be used for verification purposes.

> too little to pay for verification

-504?

> insufficient to pay for both network and sysfee combined

-511?

Reply:
> your suggestion is -500 and clarify with data?

Maybe even -32602, because transaction can't be decoded. It's not that far away from `0xgarbage` input for contract hash, for example.

Reply:
I think this one is also resolved, because everything is correct with the codes here.